<launch>
    <arg name="map_size_x" value="200.0"/>
    <arg name="map_size_y" value="40.0"/>
    <arg name="map_size_z" value=" 5.0"/>
    <arg name="odom_topic" value="/Odometry" />
    <arg name="flight_type" value="1"/>
    <arg name="drone_id" value="0"/>


  
    <include file="$(find ego_planner)/launch/include/advanced_param_exp.xml">
        <arg name="drone_id" value="$(arg drone_id)"/>
        <arg name="map_size_x_" value="$(arg map_size_x)"/>
        <arg name="map_size_y_" value="$(arg map_size_y)"/>
        <arg name="map_size_z_" value="$(arg map_size_z)"/>
        <arg name="odometry_topic" value="$(arg odom_topic)"/>
        <!-- camera pose: transform of camera frame in the world frame -->
        <!-- depth topic: depth image, 640x480 by default -->
        <!-- don't set cloud_topic if you already set these ones! -->
        <arg name="camera_pose_topic" value="nouse1"/>
        <arg name="depth_topic" value="nouse2"/>
        <!-- topic of point cloud measurement, such as from LIDAR  -->
        <!-- don't set camera pose and depth, if you already set this one! -->
        <arg name="cloud_topic" value="/cloud_registered"/>
        <!-- intrinsic params of the depth camera -->
        <arg name="cx" value="427.1639404296875"/>
        <arg name="cy" value="236.4175567626953"/>
        <arg name="fx" value="424.718017578125"/>
        <arg name="fy" value="424.718017578125"/>
        <!-- maximum velocity, acceleration and jerk the drone will reach -->
        <arg name="max_vel" value="5.0" />
        <arg name="max_acc" value="10.0" />
        <arg name="max_jer" value="15.0" />
        <!--always set to 1.5 times grater than sensing horizen-->
        <arg name="planning_horizon" value="25.0" />
        <arg name="use_multitopology_trajs" value="false" />
        <!-- 1: use 2D Nav Goal to select goal  -->
        <!-- 2: use global waypoints below  -->
        <arg name="flight_type" value="$(arg flight_type)" />
        <!-- global waypoints -->
        <!-- It generates a piecewise min-snap traj passing all waypoints -->
        <arg name="point_num" value="1" />
        <arg name="point0_x" value="15" />
        <arg name="point0_y" value="0" />
        <arg name="point0_z" value="1" />
        <arg name="point1_x" value="0.0" />
        <arg name="point1_y" value="0.0" />
        <arg name="point1_z" value="1.0" />
        <arg name="point2_x" value="15.0" />
        <arg name="point2_y" value="0.0" />
        <arg name="point2_z" value="1.0" />
        <arg name="point3_x" value="0.0" />
        <arg name="point3_y" value="0.0" />
        <arg name="point3_z" value="1.0" />
        <arg name="point4_x" value="15.0" />
        <arg name="point4_y" value="0.0" />
        <arg name="point4_z" value="1.0" />
    </include>

    <node pkg="ego_planner" name="drone_$(arg drone_id)_traj_server" type="traj_server" output="screen">
        <!--<remap from="position_cmd" to="drone_$(arg drone_id)_planning/pos_cmd"/>-->
        <remap from="~planning/trajectory" to="drone_$(arg drone_id)_planning/trajectory"/>
        <param name="traj_server/time_forward" value="1.0" type="double"/>
        
        <param name="traj_server/max_vel" value="5.0" type="double"/>
        <param name="traj_server/max_acc" value="10.0" type="double"/>
        <param name="traj_server/use_time_reallocation" value="true" type="bool"/>
        <param name="traj_server/waypoint_spacing" value="1.0" type="double"/>
        <param name="traj_server/pub_origin_cmd" value="true" type="bool"/>
    </node>

</launch>




